### PR TITLE
Moving ESA parse - compare json tests to OL

### PR DESCRIPTION
**Pull request by gjwatts**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

Borrows heavily from Chuck's PR #29273

### CODE REVIEW

**Comment by jhanders34** [comment]
#libby

File: dev/com.ibm.ws.repository.parsers/bnd.bnd [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #*******************************************************************************
-# Copyright (c) 2018 IBM Corporation and others.
+# Copyright (c) 2024 IBM Corporation and others.
```
Comment (jhanders34) [inline dev/com.ibm.ws.repository.parsers/bnd.bnd]:
```suggestion
# Copyright (c) 2018, 2024 IBM Corporation and others.
```

File: dev/com.ibm.ws.repository.parsers/build.gradle [hunk 2]
```diff
@@ -0,0 +1,84 @@
+configurations {
```
Comment (jhanders34) [inline dev/com.ibm.ws.repository.parsers/build.gradle:15]:
```suggestion
/*******************************************************************************
 * Copyright (c) 2024 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *******************************************************************************/
configurations {
```

File: dev/com.ibm.ws.repository.parsers/test/com/ibm/ws/repository/parsers/internal/CreateJsonRepositoryFilesTest.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/com.ibm.ws.repository.parsers/test/com/ibm/ws/repository/parsers/internal/CreateJsonRepositoryFilesTest.java]:
```suggestion
```
Not needed in new files.

File: dev/com.ibm.ws.repository.test.utils/bnd.bnd [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #*******************************************************************************
-# Copyright (c) 2018 IBM Corporation and others.
+# Copyright (c) 2024 IBM Corporation and others.
```
Comment (jhanders34) [inline dev/com.ibm.ws.repository.test.utils/bnd.bnd]:
```suggestion
# Copyright (c) 2018, 2024 IBM Corporation and others.
```

File: dev/com.ibm.ws.repository.test.utils/src/com/ibm/ws/repository/base/JsonAnyOrderMatcher.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/com.ibm.ws.repository.test.utils/src/com/ibm/ws/repository/base/JsonAnyOrderMatcher.java]:
```suggestion
```

File: dev/com.ibm.ws.repository.parsers/build.gradle [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    anno_metadata 'com.ibm.ws.repository.parsers.test:anno-metadata:1.0.0'
+    jsp_metadata  'com.ibm.ws.repository.parsers.test:jsp-metadata:2.3.0'
+    
+ }
+
+task copyFirstAnnoEsaFileForUnittest(type:Copy) {
+    from configurations.anno_esa
+    into new File('test-resources/esa/')
+    rename 'anno-esa-*.*.*.*', 'com.ibm.websphere.appserver.anno-1.0.esa'
+}
```
Comment (jhanders34) [inline dev/com.ibm.ws.repository.parsers/build.gradle]:
```suggestion
task copyFilesForUnittest {
  copy {
    from configurations.anno_esa
    into new File('test-resources/esa/')
    rename 'anno-esa-*.*.*.*', 'com.ibm.websphere.appserver.anno-1.0.esa'
  }
```
Update the other tasks to just be `copy {` inside of the task and the you only need to depend on one task.

File: dev/com.ibm.ws.repository.parsers/test/com/ibm/ws/repository/parsers/internal/CreateJsonRepositoryFilesTest.java (hunk 3 above)
Comment (gjwatts) [reply dev/com.ibm.ws.repository.parsers/test/com/ibm/ws/repository/parsers/internal/CreateJsonRepositoryFilesTest.java]: I did not know that, thanks for pointing that out.

File: dev/com.ibm.ws.repository.parsers/build.gradle (hunk 6 above)
Comment (gjwatts) [reply dev/com.ibm.ws.repository.parsers/build.gradle]: Thank you!  That looks much cleaner.

**Comment by gjwatts** [comment]
#build

**Review by jhanders34 — APPROVED** [review]

**Comment by gjwatts** [comment]
Only 2 failures with an unrelated UI component.

**Review by cbridgha — APPROVED** [review]
Looks good!

**Pull request merged**